index.js

```diff
@@ -44,6 +44,52 @@ function fastifyView (fastify, opts, next) {
     return page
   }
 
+  // Gets template as string from LRU cache or filesystem.
+  const getTemplateString = function (file, callback) {
+    let data = lru.get(file)
+    if (data && prod) {
+      callback(null, data)
+    } else {
+      readFile(join(templatesDir, file), 'utf-8', (err, data) => {
+        if (err) {
+          callback(err, null)
+          return
+        }
+        lru.set(file, data)
+        callback(null, data)
+      })
+    }
+  }
+
+  // Gets partials as collection of strings from LRU cache or filesystem.
+  const getPartials = function (page, partials, callback) {
+    let partialsObj = lru.get(`${page}-Partials`)
+
+    if (partialsObj && prod) {
+      callback(null, partialsObj)
+    } else {
+      let filesToLoad = Object.keys(partials).length
+
+      if (filesToLoad === 0) {
+        callback(null, {})
+      }
+
+      Object.keys(partials).map((key, index) => {
+        readFile(join(templatesDir, partials[key]), 'utf-8', (err, data) => {
+          if (err) {
+            callback(err, null)
```

Comment:
If one of the read errors, there is a high chance of the callback being called more than once.